src/SAWScript/Crucible/MIR/ResolveSetupValue.hs

```diff
@@ -312,48 +362,198 @@ resolveSetupVal mcc env tyenv nameEnv val =
+              -- a symbolic bitvector from it.
+              Some discrTpr <- pure $ Mir.tyToRepr col discrTp
+              let discrShp = tyToShapeEq col discrTp discrTpr
+              IsBVShape _ discrW <- pure $ testDiscriminantIsBV discrShp
+              let discr = getEnumVariantDiscr variant
+              discrVal <- W4.bvLit sym discrW $ BV.mkBV discrW discr
+
+              -- Construct an EnumShape and RustEnumRepr. This requires
+              -- processing /all/ variants, not just the particular variant that
+              -- we are building.
+              (enumShp, Some expectedVariantShps) <- pure $
+                enumShapes adt discrTp discrShp variants
+              let variantTprs =
+                    FC.fmapFC variantShapeType expectedVariantShps
+              let enumTpr = Mir.RustEnumRepr
+                              discrTpr
+                              variantTprs
+
+              -- Construct the VariantShape of the particular variant that we
+              -- are building.
+              Some variantIdx <- pure $
+                variantIntIndex nm variantIdxInt $
+                Ctx.size expectedVariantShps
+              VariantShape expectedFldAssn <-
+                pure $ expectedVariantShps Ctx.! variantIdx
+
+              -- Check that the actual field values match the expected types.
+              Some (Functor.Pair actualFldAssn actualValAssn) <-
+                pure $ variantFieldsToAssns sym flds'
+              Refl <-
+                case W4.testEquality expectedFldAssn actualFldAssn of
+                  Just r -> pure r
+                  Nothing ->
+                    panic "resolveSetupVal"
+                          [ "Enum field shape mismatch"
+                          , "Expected: " ++ show expectedFldAssn
+                          , "Actual: " ++ show actualFldAssn
+                          ]
+
+              -- Finally, construct a MIRVal.
+              let enumVal =
+                    Ctx.empty
+                      Ctx.:> RV discrVal
+                      Ctx.:> RV (injectVariant sym variantTprs variantIdx actualValAssn)
+              pure $ MIRVal enumShp $ AnyValue enumTpr enumVal
+            Mir.Adt nm Mir.Struct _ _ _ _ _ ->
+              panic "resolveSetupVal"
+                    [ "Expected enum type, received struct:"
+                    , show nm
+                    ]
+            Mir.Adt nm Mir.Union _ _ _ _ _ ->
+              panic "resolveSetupVal"
+                    [ "Expected enum type, received union:"
+                    , show nm
+                    ]
+        -- See Note [Symbolic enums] in SAWScript.Crucible.MIR.Setup.Value for
+        -- more information on the approach used to resolve symbolic enum
+        -- values.
+        MirSetupEnumSymbolic adt discr variantFlds ->
+          case adt of
+            Mir.Adt nm (Mir.Enum discrTp) variants _ _ _ _ -> do
```

Comment:
Good point. For what it's worth, due to how `mir_fresh_expanded_value` is implemented, I don't think it is ever possible to construct a `MirSetupEnumSymbolic adt` value where the `adt` is `repr(transparent)`. Nevertheless, it could be helpful to check this invariant explicitly, so I've added a `repr(transparent)` check to be on the safe side. I've also made a note of this invariant in the Haddocks for `MirSetupEnumSymbolic`.